src/TimeSpans.jl

```diff
@@ -43,6 +43,17 @@ Return `TimeSpan(start(x), stop(x))`.
 """
 TimeSpan(x) = TimeSpan(start(x), stop(x))
 
+function Base.parse(::Type{TimeSpan}, str::AbstractString)
```

Comment:
This implies that we've decided on a canonical string representation for a `TimeSpan`?

This looks like it's parsing a PostgreSQL integer range, right? I'm not sure if it's a good idea (or not, maybe it is?) to formally couple TimeSpans to PostgreSQL integer ranges, seems a bit like a leaked abstraction...if this is to avoid type piracy in internal Beacon code, than IMO it'd be better for that internal Beacon code to have this snippet in a utility function (`postgres_int_range_to_timespan` or something) vs. punning on an overload of `Base.parse`.

Reply:
Also note that as written, this wouldn't recover a `TimeSpan` from its `show` output.

Reply:
All valid points. LibPQ.jl has it's own parsing function called `pqparse`. We don't need to extend that function but it seems sensible to re-use the name?

Reply:
>  IMO it'd be better for that internal Beacon code to have this snippet in a utility function (`postgres_int_range_to_timespan` or something) vs. punning on an overload of `Base.parse`.

I think this is the only relevant concern left. I would like to avoid the type-piracy but this can be internalized. I can move the tests over to the internal package if that is the overall preference.

Reply:
Yeah, there's no reason for TimeSpans.jl to maintain a downstream caller's PostgreSQL interaction code, IMO. My vote is we move this PR there

Reply:
Will do. If I find additional cases I may end up pushing for an extension package in the future.